container/container.go

```diff
@@ -484,7 +484,9 @@ func copyEscapable(dst io.Writer, src io.ReadCloser, keys []byte) (written int64
 		nr, er := src.Read(buf)
 		if nr > 0 {
 			// ---- Docker addition
+			preservBuf := []byte{}
```

Comment:
@duglin hum it matches a part of the key, true, until the last *wrong* key is entered. Could have done something with `keys`, `i` and the latest read but it failed a little less readable to me :sweat_smile: